test/junit/scala/tools/nsc/classpath/ZipAndJarFileLookupFactoryTest.scala

```diff
@@ -29,21 +29,45 @@ class ZipAndJarFileLookupFactoryTest {
       // check things work after the cache hit
       cp1.findClassFile("p2.X").get.toByteArray
 
+      // Change only one letter of contents to force regeneration via a CRC change
+      createZip(f, "not empty".getBytes, "p1/C.class")
+      val cp3 = createCp
+      assert(cp3.findClass("p1.C").isDefined)
+      assert(cp3.findClass("p2.X").isDefined)
+      assert(cp3.findClass("p3.Y").isDefined)
+      assert(cp1 ne cp3, (System.identityHashCode(cp1), System.identityHashCode(cp3)))
+
       val lastMod1 = Files.getLastModifiedTime(f)
       // Create a new zip at the same path with different contents and last modified
       Files.delete(f)
       createZip(f, Array(), "p1/D.class")
       Files.setLastModifiedTime(f, FileTime.fromMillis(lastMod1.toMillis + 2000))
 
       // Our classpath cache should create a new instance
-      val cp3 = createCp
-      assert(cp1 ne cp3, (System.identityHashCode(cp1), System.identityHashCode(cp3)))
+      val cp4 = createCp
+      assert(cp3 ne cp4, (System.identityHashCode(cp3), System.identityHashCode(cp4)))
       // And that instance should see D, not C, in package p1.
-      assert(cp3.findClass("p1.C").isEmpty)
-      assert(cp3.findClass("p1.D").isDefined)
+      assert(cp4.findClass("p1.C").isEmpty)
+      assert(cp4.findClass("p1.D").isDefined)
     } finally Files.delete(f)
   }
 
+  @Test def createClasspathFor50Jars(): Unit = {
```

Comment:
I don't think we should commit this test, although I'm happy you're giving performance a sanity check.